api/lib/domain/models/CorrectionBlockQROCMDep.js

```diff
@@ -0,0 +1,63 @@
+const Joi = require('joi');
+
+class CorrectionBlockQROCMDep {
+  /**
+   * @type {boolean}
+   */
+  #validated;
+
+  /**
+   * @type {string[]}
+   */
+  #alternativeSolutions;
+
+  /**
+   * @param {boolean} validated
+   * @param {string[]} alternativeSolutions
+   */
+  constructor(validated = false, alternativeSolutions = []) {
+    this.#validateValidatedArgument(validated);
+    this.#validateAlternativeSolutionsArgument(alternativeSolutions);
+
+    this.#validated = validated;
+    this.#alternativeSolutions = alternativeSolutions;
+  }
```

Comment:
J'ai bien vu que cette PR était close, mais je n'ai pas pu mettre ce commentaire que je voulais mettre avant, désolé !

Dans l'implémentation qui a été mergée je trouve dommage de faire de la validation sur des valeurs par  défaut d'arguments, c'est une dépense d'énergie inutile. Aussi je propose ci-dessous une implémentation : 
* qui ne valide `alternativeSolutions` que lorsque ce paramètre est présent, le membre `#alternativeSolutions` ayant une valeur par défaut au niveau de sa définition dans la classe. Dans le cas de cette classe ce n'est pas grave de valider inutilement 1 paramètre, mais ce n'est pas un comportement qui monte à l'échelle et donc pas à encourager
* qui rend le paramètre `validated` obligatoire car dans l'usage dans le service on voit qu'on a envie de mettre `true` ou `false`

Enfin, note additionnelle, si on utilise JSDoc on peut tirer partie de la notation qui indique que certains paramètres de fonctions/méthodes sont optionnels, ce sont les crochets qui sont ajoutés dans `[alternativeSolutions]` dans la JSDoc du constructeur cf. https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

```javascript
  #alternativeSolutions = [];

  /**
   * @param {boolean} validated
   * @param {string[]} [alternativeSolutions]
   */
  constructor(validated, alternativeSolutions) {
    this.#validateValidatedArgument(validated);
    this.#validated = validated;

    if (alternativeSolutions) {
      this.#validateAlternativeSolutionsArgument(alternativeSolutions);
      this.#alternativeSolutions = alternativeSolutions;
    }
  }
```

Reply:
Je suis complètement d'accord, ca va tout a fait dans le sens de ce qu'on a essayé de faire. Je vais modifier le model dans une PR suivante.